### PR TITLE
fix: resolve conflict

### PR DESCRIPTION
**Pull request by evilebottnawi**

### CODE REVIEW

File: lib/webpackImporter.js [hunk 1]
```diff
@@ -46,6 +46,9 @@ function webpackImporter(resourcePath, resolve, addNormalizedDependency) {
             Promise.reject() :
             resolve(dir, importsToResolve[0])
                 .then(resolvedFile => {
+                    if (!([".sass", ".scss", ".css"].includes(path.extname(resolvedFile)))) {
+                        throw new Error("Alias should have 'sass', 'scss' or 'css' extension");
```
Comment (jhnns) [inline lib/webpackImporter.js]: This requirement is very arbitrary and I don't think that we should introduce that. Why do aliases have to have a `sass`, `scss` or `css` extension? Furthermore, it's not necessarily an alias that caused this condition, so the error message is also misleading.

Comment (evilebottnawi) [reply lib/webpackImporter.js]: @jhnns i know :disappointed: , it is very interesting. Found solution:
1.  Add here url + all paths from `includePaths` https://github.com/webpack-contrib/sass-loader/blob/master/lib/importsToResolve.js#L56 (i.e. `includePath[0]/${url}`, `includePath[1]/${url}`, url)
2. Node-sass should be try import first when we try import throw own implement (not sure it is possible)

Any ideas how we can fix this?

Comment (jhnns) [reply lib/webpackImporter.js]: Currently not. I wish we could let node-sass handle the imports as default and then use webpack's resolver, but that's not possible with the current node-sass API.

Comment (evilebottnawi) [reply lib/webpackImporter.js]: @jhnns :disappointed: I try to search solution todat, have some ideas.

**Comment by jhnns** [comment]
This change introduces other problems as described [here](https://github.com/webpack-contrib/sass-loader/pull/558#discussion_r183700457).

I think we need to go a similar route as with the less-loader here. If the `includePaths` option is specified, webpack's resolver isn't used at all (see [less-loader](https://github.com/webpack-contrib/less-loader#imports)). It makes no sense to specify `includePaths` *and* to use webpack's resolver (because webpack's resolver provides the same possibilities as `includePaths` do).

Actually, I'm even thinking about making webpack's resolver opt-in because I assume that the  threadpool issue with custom importers #147 is the cause for the performance problems #296. Furthermore, most people don't need webpack's resolver. They just want to use Sass the regular way.

If the webpack resolver is used, the `includePaths` option is not valid anymore.

**Comment by evilebottnawi** [comment]
@jhnns what about don't resolve path includes `includePaths`?

**Comment by jhnns** [comment]
The problem is that we can only know after resolving that an import was supposed to be resolved by `includePaths`. Besides that, I think it makes sense to make the webpack resolver optional.

**Comment by evilebottnawi** [comment]
@jhnns `optional` solve problem, but many new people will be set this `optional` to `true` and get errors - create issue, i think better will be solve problem without new option to avoid this situation. Will you give me time to think more deeply about this?

**Comment by evilebottnawi** [comment]
@jhnns i faced with problem in some project i need alias and `includePaths`, i think `optional` is not real solve problem

**Pull request closed without merging**